### PR TITLE
[20542] Adapt nightly jobs for all supported versions

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR removes the versioning in the .repos workflow configuration files to let each version contain its proper .repos file.
This PR would be cherry picked in the supported versions.

Related PRs:

* 2.6.x: #4484
* 2.10.x: #4556
* 2.12.x: #4482
* 2.13.x: #4481

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A**  Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A**  Any new/modified methods have been properly documented using Doxygen. 
- **N/A**  Changes are ABI compatible. 
- **N/A**  Changes are API compatible. 
- **N/A**  New feature has been added to the `versions.md` file (if applicable).
- **N/A**  New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [x] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: .github/workflows/config/ci_colcon.meta
Comment (EduPonz) [inline .github/workflows/config/ci_colcon.meta:1]: I'd suggest ci.meta

File: .github/workflows/reusable-mac-ci.yml [hunk 1]
```diff
@@ -85,35 +85,21 @@ jobs:
...
           cp ${META_PATH} ci.meta
           cat ci.meta
 
-          # Create source dir and download the sources
-          vcs import src --input ${DEPENDS_REPOS_PATH} --skip-existing
+          # Create source dir and download the version sources
+          vcs import src --input ./src/fastrtps/fastrtps.repos --skip-existing
+
+          # Download the CI sources
+          vcs import src --input ${REPOS_PATH} --skip-existing
```
Comment (EduPonz) [inline .github/workflows/reusable-mac-ci.yml:102]: I'd transform this into 2 calls to eprosima-CI action. Mind that you'dd need to adjust the path to the meta file on the build step. Same for all workflows

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-mac-ci.yml [hunk 2]
```diff
@@ -109,7 +104,7 @@ jobs:
         continue-on-error: false
         uses: eProsima/eProsima-CI/multiplatform/colcon_build@v0
         with:
-          colcon_meta_file: ${{ github.workspace }}/ci.meta
+          colcon_meta_file: ./src/fastrtps/.github/workflows/config/ci.meta
```
Comment (EduPonz) [inline .github/workflows/reusable-mac-ci.yml:107]: I think it's safer to keep abs paths everywhere. Applies to all changes
```suggestion
          colcon_meta_file: ${{ github.workspace }}/src/fastrtps/.github/workflows/config/ci.meta
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Comment by EduPonz** [comment]
@JesusPoderoso I think we need to run CI here

**Review by JesusPoderoso — COMMENTED** [review]
LGTM with green CI

**Review by EduPonz — APPROVED** [review]

**Pull request merged**